napi_basics.md

```diff
@@ -0,0 +1,120 @@
+to manage the lifetimes of JavaScript values, including defining their minimum
+lifetimes explicitly, and also being notified when they are garbage-collected.
+
+### *napi_handle_scope*
+This is an abstraction used to provide hints to the garbage collector about the location of 
+handles to GC pointers. napi_handle_scope allows the user to provide a variable stack
+separate from the thread stack, in a manner that is efficient for scan in case 
+the underlying VM uses a garbage collector with precise-semantics.
+N-API values are created within the context of a handle scope- if the user does not 
+create a handle scope, they are created within the default handle scope. In the case of a 
+libuv callback, there is no default handle scope so the user is required to supply one.
+Handle scopes are created using `napi_open_handle_scope` and are destroyed using 
+`napi_close_handle_scope`. Calling close can indicate to the GC that all napi_values 
+created during the lifetime of the handle scope are no longer referenced from the
+current stack frame.
+
+Consider creating a `napi_handle_scope` per function if you create any handle other than 
+just the return value. Otherwise, the handle created will leak to the parent handle scope
+which can lead to higher than expected (and necessary) memory usage.
+
+### *napi_escapable_handle_scope*
+Escapable Handle Scopes are a special type of handle scope that are used to return
+values created within a particular handle scope to a parent scope.
+
+### *napi_callback_info*
+Opaque datatype that is passed to a callback function. It can be used for two purposes:
+- Get additional information about the context in which the callback was invoked
+- Set the return value of the callback
+
+### *napi_callback*
+Function pointer type for N-API APIs that call back into user-provided native code.
+Your callback function should satisfy the following signature:
+```
+typedef void (*napi_callback)(napi_env, napi_callback_info);
+```
+
+### *napi_finalize*
+Function pointer type for N-API APIs that allows the user to be notified when externally-owned data is ready to be cleaned up becauuse the object that it was associated with is garbage collected. The user must provide a function satisfying the following signature which would get called upon the object's collection.
+```
+typedef void (*napi_finalize)(void* finalize_data, void* finalize_hint);
+```
+
+### *napi_status*
+Integral status code indicating the success or failure of a N-API API call.
+Currently, the following status codes are supported.
+```
+  napi_ok,
+  napi_invalid_arg,
+  napi_object_expected,
+  napi_string_expected,
+  napi_function_expected,
+  napi_number_expected,
+  napi_boolean_expected,
+  napi_generic_failure,
+  napi_pending_exception,
+  napi_status_last
+```
+If additional information is required upon an API returning a failed status, it can
+be obtained by calling `napi_get_last_error_info`.
+
```

Comment:
Took the first suggestion. Tried to take the second, let me know if what I did was what you had in mind.